include/igl/cr_vector_laplacian.cpp

```diff
@@ -0,0 +1,128 @@
+// This file is part of libigl, a simple c++ geometry processing library.
+//
+// Copyright (C) 2020 Oded Stein <oded.stein@columbia.edu>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can
+// obtain one at http://mozilla.org/MPL/2.0/.
+#include "cr_vector_laplacian.h"
+
+#include <vector>
+
+#include "orient_halfedges.h"
+
+#include "doublearea.h"
+#include "squared_edge_lengths.h"
+
+
+template <typename DerivedV, typename DerivedF, typename DerivedE,
+typename DerivedOE, typename ScalarL>
+IGL_INLINE void
+igl::cr_vector_laplacian(
+ const Eigen::MatrixBase<DerivedV>& V,
+ const Eigen::MatrixBase<DerivedF>& F,
+ const Eigen::MatrixBase<DerivedE>& E,
+ const Eigen::MatrixBase<DerivedOE>& oE,
+ Eigen::SparseMatrix<ScalarL>& L)
+{
+  Eigen::Matrix<typename DerivedV::Scalar, Eigen::Dynamic, Eigen::Dynamic>
+  l_sq;
+  squared_edge_lengths(V, F, l_sq);
+  cr_vector_laplacian_intrinsic(F, l_sq, E, oE, L);
+}
+
+
+template <typename DerivedV, typename DerivedF, typename DerivedE,
+typename DerivedOE, typename ScalarL>
+IGL_INLINE void
+igl::cr_vector_laplacian(
+ const Eigen::MatrixBase<DerivedV>& V,
+ const Eigen::MatrixBase<DerivedF>& F,
+ Eigen::PlainObjectBase<DerivedE>& E,
+ Eigen::PlainObjectBase<DerivedOE>& oE,
+ Eigen::SparseMatrix<ScalarL>& L)
+{
+  if(E.rows()!=F.rows() || E.cols()!=F.cols() || oE.rows()!=F.rows() ||
+   oE.cols()!=F.cols())
+    orient_halfedges(F, E, oE);
+
+  cr_vector_laplacian(V, F,
+    const_cast<const Eigen::PlainObjectBase<DerivedE>& >(E),
```

Comment:
`const_cast` is not needed to add constness, only to remove it. You can use the following instead:
```cpp
const Eigen::PlainObjectBase<DerivedE>& E_cst(E);
const Eigen::PlainObjectBase<DerivedOE>& oE_cst(oE);
cr_vector_laplacian(E_cst, oE_cst, L);
```